.github/workflows/tests.yaml

```diff
@@ -58,8 +58,12 @@ jobs:
             GO_BUILD_FLAGS='-v' GOARCH=s390x ./build
             ;;
           linux-amd64-grpcproxy)
+            set -o pipefail; set +e;
             PASSES='build grpcproxy' CPU='4' RACE='true' ./test 2>&1 | tee test.log
-            ! egrep "(--- FAIL:|DATA RACE|panic: test timed out|appears to have leaked)" -B50 -A10 test.log
+            ecode=$?
+            egrep "(--- FAIL:|DATA RACE|panic: test timed out|appears to have leaked)" -B50 -A10 test.log
```

Comment:
Why do you remove the `!` ? We may need to keep it. If we find any `(--- FAIL:|DATA RACE|panic: test timed out|appears to have leaked)`, we may still consider the test failed.

Reply:
Sorry for the late reply.

 I was thinking if we can delete the `egrep` command in this patch. 

I didn't see any comment about egrep introduced in https://github.com/etcd-io/etcd/pull/8655. 
It seems that the egrep is used to fix the unexpected exit code, since `| tee` won't return expected exit code because of `pipefail`.

As the `tee` already writes the log in the stdout, we won't lost any details and the command will return expected exit code. Maybe we can go with the following script:

```bash
set -o pipefail
PASSES='build grpcproxy' CPU='4' RACE='true' ./test 2>&1 | tee test.log
```

Does it make senses?

Reply:
It should be OK, because the following case will never happen:
The test script/cases exits with 0, but somehow there are some `(--- FAIL:|DATA RACE|panic: test timed out|appears to have leaked)` errors.

Reply:
Yes. So I believe that egrep is to fix the `pipefail` issue. I will update the pr later and send new pr to main branch as well.

Reply:
One of the things that first caught my eye in etcd scripting is fact that it doesn't run in strict mode and it even depends on some parts failing for correctness. By strict mode I mean `set -euo pipefail` as defined in http://redsymbol.net/articles/unofficial-bash-strict-mode/. 

To my understanding `!grep (--- FAIL:|...` was a workaround for not properly written scripts. I think long term it would be better to fix all scripts and switch on strict mode. What do you think? @ahrtr @fuweid

Reply:
> By strict mode I mean `set -euo pipefail` as defined in

Makes sense to me.